processing/src/main/java/io/druid/segment/serde/DictionaryEncodedColumnPartSerde.java

```diff
@@ -83,7 +87,169 @@ public byte asByte()
+    {
+      this.spatialIndexWriter = spatialIndexWriter;
+      return this;
+    }
+
+    public SerializerBuilder withByteOrder(ByteOrder byteOrder)
+    {
+      this.byteOrder = byteOrder;
+      return this;
+    }
+
+    public SerializerBuilder withValue(IndexedIntsWriter valueWriter, boolean hasMultiValue, boolean compressed)
+    {
+      this.valueWriter = valueWriter;
+      if (hasMultiValue) {
+        if (compressed) {
+          this.version = VERSION.COMPRESSED;
+          this.flags |= Feature.MULTI_VALUE_V3.getMask();
+        } else {
+          this.version = VERSION.UNCOMPRESSED_MULTI_VALUE;
+          this.flags |= Feature.MULTI_VALUE.getMask();
+        }
+      } else {
+        if (compressed) {
+          this.version = VERSION.COMPRESSED;
+        } else {
+          this.version = VERSION.UNCOMPRESSED_SINGLE_VALUE;
+        }
+      }
+      return this;
+    }
+
+    public DictionaryEncodedColumnPartSerde build()
+    {
+      return new DictionaryEncodedColumnPartSerde(
+          byteOrder,
+          bitmapSerdeFactory,
+          new Serializer()
+          {
+            @Override
+            public long numBytes()
+            {
+              long size = 1 + // version
+                          (version.compareTo(VERSION.COMPRESSED) >= 0
+                           ? Ints.BYTES
+                           : 0); // flag if version >= compressed
+              if (dictionaryWriter != null) {
+                size += dictionaryWriter.getSerializedSize();
+              }
+              if (valueWriter != null) {
+                size += valueWriter.getSerializedSize();
+              }
+              if (bitmapIndexWriter != null) {
+                size += bitmapIndexWriter.getSerializedSize();
+              }
+              if (spatialIndexWriter != null) {
+                size += spatialIndexWriter.getSerializedSize();
+              }
+              return size;
+            }
+
+            @Override
+            public void write(WritableByteChannel channel) throws IOException
+            {
+              channel.write(ByteBuffer.wrap(new byte[]{version.asByte()}));
+              if (version.compareTo(VERSION.COMPRESSED) >= 0) {
+                channel.write(ByteBuffer.wrap(Ints.toByteArray(flags)));
+              }
+              if (dictionaryWriter != null) {
+                dictionaryWriter.writeToChannel(channel);
+              }
+              if (valueWriter != null) {
+                valueWriter.writeToChannel(channel);
+              }
+              if (bitmapIndexWriter != null) {
+                bitmapIndexWriter.writeToChannel(channel);
+              }
+              if (spatialIndexWriter != null) {
+                spatialIndexWriter.writeToChannel(channel);
+              }
+            }
+          }
+      );
+    }
+  }
+
+  public static LegacySerializerBuilder legacySerializerBuilder()
```

Comment:
any advises for naming the version? both serializer are suitable for v9 index, so we can not call the legacy V8 and the new one V9